lib/coordinate_validation.rb

```diff
@@ -0,0 +1,33 @@
+# This module defines the valid battleship coordinates using regex.
+# This can be called in any class where this should be validated.
+# The project instructions defines the grid as the following:
+#    A   B   C   D
+# 1 |_| |_| |_| |_|
+# 2 |_| |_| |_| |_|
+# 3 |_| |_| |_| |_|
+# 4 |_| |_| |_| |_|
+
+module CoordinateValidation
+  # This is a constant that defines the valid battleship coordinates using regex.
+  # The regex means that the coordinate must start with a letter A-D, and end with a number 1-4.
+  VALID_COORDINATE_REGEX = /^[A-Z]{1}\d{1,2}$/
```

Comment:
Great work diving into regular expressions! Regex is an often loathed thing in software development. But you tackled it nicely!